File: adapterlib/src/main/java/com/hxz/adapterlib/viewholder/CloserViewHolder.java
package com.hxz.adapterlib.viewholder;

import android.util.SparseArray;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @author Closer
 * @deprecated 通用 holder
 */
public class CloserViewHolder extends RecyclerView.ViewHolder {

    // 布局中子view缓存
    private SparseArray<View> mViews;

    // 根布局
    private View mConvertView;


    public CloserViewHolder(@NonNull View itemView) {
        super(itemView);
        mConvertView = itemView;
        mViews = new SparseArray<>();
    }


    /**
     * 绑定item view布局，生成view holder
     */
    public static CloserViewHolder bindView(ViewGroup parent, int layoutId) {
        View view = LayoutInflater.from(parent.getContext()).inflate(layoutId, parent, false);
        return new CloserViewHolder(view);
    }

    /**
     * 通过id 获取item view中的view 控件
     *
     * @param <T> 控件类型
     */
    public <T extends View> T getView(int viewId) {
        View view = mViews.get(viewId);
        if (view == null) {
            view = mConvertView.findViewById(viewId);
            mViews.put(viewId, view);
        }
        return (T) view;
    }


}

File: adapterlib/src/main/java/com/hxz/adapterlib/kt/CloserViewHolder.kt
package com.hxz.adapterlib.kt

import android.util.SparseArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 * @Author:andy
 * @CreateDate:2020/9/19 17:09
 * @Description:改写Kotlin,通用ViewHolder
 */
class CloserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    //存放View
    private var mViews: SparseArray<View>? = null

    //根View
    private var mRootView: View? = null


    init {
        mRootView = itemView
        mViews = SparseArray()
    }

    /**
     * 绑定item view布局，生成view holder
     */
    fun bindView(mViewGroup: ViewGroup, mLayoutId: Int): CloserViewHolder? {

        return CloserViewHolder(LayoutInflater.from(mViewGroup.context).inflate(mLayoutId, mViewGroup, false))
    }


    /**
     *  通过id 获取item view中的view 控件
     */
    fun <T : View?> getView(mViewId: Int): T {
        var mView = mViews!![mViewId]
        if (mView == null) {
            mView = mRootView!!.findViewById(mViewId)
            mViews!!.put(mViewId, mView)
        }
        return mView as T
    }


}